source/Ship.cpp

```diff
@@ -3536,6 +3536,7 @@ void Ship::UnloadBays()
 			carriedMass -= bay.ship->Mass();
 			bay.ship->SetSystem(currentSystem);
 			bay.ship->SetPlanet(landingPlanet);
+			bay.ship->UnmarkForRemoval();
```

Comment:
Not sure why this is required?

Reply:
There is a nice [analysis by Vitalchip](https://github.com/endless-sky/endless-sky/issues/6519#issuecomment-1025215292) in the reporting issue on why this is required.

Reply:
> Addressing point 3 - Deploying and recalling the Finch sets "shouldRemove" but this isn't unset when unloading bays, only when launching.

Seems to me lile we should instead unset it when unloading bays but I dont know this code in details so take my advice with a grain of salt.

Reply:
Well, this is in the `Ship::UnloadBays()` function, so it looks like we are now actually doing exactly what you suggest.

We might still be able to optimize if we take the common code from Launch() and from UnloadBays() and place such common code on 1 single location. But let's keep that for another PR.